src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  i;
+  static alive = [];
+
+  constructor(name, health = FULL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = FULL_HEALTH) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= BITE_DAMAGE;
+      herbivore.die();
```

Comment:
[CODE PERFORMANCE] - After biting and potentially reducing the health of a `Herbivore`, the `die` method is called which filters the `Animal.alive` array. However, this could lead to performance issues if called frequently on many animals. It's recommended to check the health of the herbivore before filtering the `Animal.alive` array to avoid unnecessary filtering when the health is above 0.